bundle/edu.gemini.smartgcal.servlet/src/main/java/edu/gemini/smartgcal/servlet/repository/CalibrationSubversionRepository.java

```diff
@@ -40,6 +33,33 @@ public CalibrationSubversionRepository(String url, String user, String password)
         if (url      == null) throw new IllegalArgumentException("url must not be null");
         if (user     == null) throw new IllegalArgumentException("user must not be null");
         if (password == null) throw new IllegalArgumentException("password must not be null");
+
+        // If svn is not available CmdLineClientAdapterFactory.setup() will fail with an exception.
+        // Unfortunately the original exception is swallowed and the one thrown by setup() does not have a
+        // case and is generic and therefore gives no clues as to what went wrong. This additional code tries
+        // to execute "svn --version" manually (which is what the setup() function is doing) and log the exception,
+        // hopefully giving us some additional information.
+        try {
+            LOG.info("Trying to execute svn --version to validate svn availability");
+            final CmdLineClientAdapter12 cmd = new CmdLineClientAdapter12(new CmdLineNotificationHandler());
+            cmd.getVersion();
+            LOG.info("svn --version: OK");
```

Comment:
I think it was a good idea to move this out of a static initializer block and to explicitly do this test before trying the setup. So :+1: let's merge this.  Having gotten the version information though, we might as well log what it is just in case it somehow finds a different version of svn than we were expecting.